configdialog.cpp

```diff
@@ -518,6 +535,271 @@ void PaletteCustomizeDialog::get_and_save_color(int palette_index) {
+			current_is_valid |= static_cast<uint8_t>(Errors::name_too_short);
+		}
+		if (current_name.length() >= PaletteCustomizeDialog::MaxPaletteNameSize) {
+			current_is_valid |= static_cast<uint8_t>(Errors::name_too_long);
+		}
+		if (current_name.find_first_not_of(valid_chars + L' ') != wstring::npos) {
+			current_is_valid |= static_cast<uint8_t>(Errors::name_invalid_characters);
+		}
+
+		current_index = palettes->find_first_of(L':', current_index) + 1;
+		for (int i = 0; i < colors_amount; i++) {
+			if (palettes->find_first_of(L':', current_index) < palettes->find_first_of(L'#', current_index)) {
+				current_is_valid |= static_cast<uint8_t>(Errors::colors_too_few);
+				break;
+			}
+
+			if (palettes->find_first_of(L'#', current_index) == wstring::npos) {
+				current_is_valid |= static_cast<uint8_t>(Errors::colors_too_few);
+				break;
+			}
+			current_index = palettes->find_first_of(L'#', current_index) + 1;
+			current_colors[i] = palettes->substr(current_index, palettes->find_first_of(L';', current_index) - current_index);
+			if (palettes->find_first_of(L';', current_index) == wstring::npos) {
+				current_is_valid |= static_cast<uint8_t>(Errors::colors_too_few);
+				break;
+			}
+			current_index = palettes->find_first_of(L';', current_index) + 1;
+
+			if (current_colors[i].length() != 6) {
+				current_is_valid |= static_cast<uint8_t>(Errors::colors_incorrect_length);
+			}
+			if (current_colors[i].find_first_not_of(valid_hex_chars) != wstring::npos) {
+				current_is_valid |= static_cast<uint8_t>(Errors::colors_invalid_characters);
+			}
+		}
+
+		if (current_is_valid != static_cast<uint8_t>(Errors::none)) {
+			count_invalid++;
+		}
+
+		names.push_back(current_name);
+		colors.push_back(current_colors);
+		is_valid.push_back(current_is_valid);
+		printf("");
+	}
+
+	bool import = true;
+	if (count_invalid <= 1 && !graceful_exit && (count_invalid == 0 || *is_valid.rbegin() != static_cast<uint8_t>(Errors::none))) {
+		wstring error_info = L"No terminating character (~) detected. One or more palettes may be missing.\nLast detected palette: " + *(names.rbegin() + count_invalid);
+		error_info += L"\n\nWould you like to import the palettes anyway?\n\nThe following palettes can be imported:";
+		for (size_t i = 0; i < palettes_amount; i++) {
+			if (is_valid[i] == static_cast<uint8_t>(Errors::none)) {
+				error_info += L" " + names[i] + L",";
+			}
+		}
+		*error_info.rbegin() = L'.';
+
+		int num = MessageBox(
+			NULL,
+			error_info.c_str(),
+			L"Palette Customizer",
+			MB_YESNO | MB_ICONWARNING | MB_DEFBUTTON2
+		);
+		import = (num == IDYES);
+	}
+	else if (count_invalid > 0) {
+		wstring error_info = L"One or more palettes have errors:\n";
+		for (size_t i = 0, invalid = 0; i < palettes_amount; i++) {
+			if (is_valid[i] != static_cast<uint8_t>(Errors::none)) {
+				invalid++;
+				wstring name = names[i];
+				while (name.find_first_of(L'\r') != wstring::npos) {
```

Comment:
This is something I do want to make sure is changed though. We cannot have newlines, they are too dangerous.